utils/build/docker/dotnet/weblog/Controllers/DebuggerController.cs

```diff
@@ -119,5 +119,40 @@ public async Task<IActionResult> Nulls(int? intValue = null, string strValue = n
             PiiBase? pii = await Task.FromResult<PiiBase>(null);
             return Content($"Pii is null {pii is null}. intValue is null {intValue is null}. strValue is null {strValue is null}.");
         }
+
+        [HttpGet("exceptionreplay/simple")]
+        [Consumes("application/json", "application/xml")]
+        public IActionResult ExceptionReplaySimple()
+        {
+            throw new System.Exception("Simple exception");
+        }
+
+        [HttpGet("exceptionreplay/recursion5")]
+        [Consumes("application/json", "application/xml")]
+        public IActionResult ExceptionReplayRecursion5(int depth = 5)
+        {
+            if (depth > 0)
+            {
+                return ExceptionReplayRecursion5(depth - 1);
+            }
+            else
+            {
+                throw new System.Exception("Recursion exception");
+            }
+        }
+
+        [HttpGet("exceptionreplay/recursion20")]
+        [Consumes("application/json", "application/xml")]
+        public IActionResult ExceptionReplayRecursion20(int depth = 20)
+        {
+            if (depth > 0)
+            {
+                return ExceptionReplayRecursion20(depth - 1);
+            }
+            else
+            {
+                throw new System.Exception("Recursion exception");
+            }
+        }
```

Comment:
Can't you simply pass the `depth` via url query param? - it will allow us to test exciting things, such as, querying varying deepness (10, 9, 8, ...) and check that each of them 'finalizes' upon second execution, and do so in various ordering. E.g requesting `exceptionreplay/recursion/10`, `exceptionreplay/recursion/9`, ... and then requesting the same endpoints in various orders to see that we submit snapshots for each of them individually and at correct times (upon second request, and not capturing any snapshots when hitting those endpoints with varying recursion deepness, etc)?

Reply:
I can, but it will be hard to find related snapshots due to system test limitations. Right now, the only way I can identify related snapshots is by the unique method name. The system tests are very robust but don't allow much flexibility.

Reply:
Would it be possible to inject identifiers in the query ?

Reply:
It possible, but again makes things even more complicated to identify. For now I created a simplest way to test this functionality. If we want to make things more complicated and cover more cases, we can do it in future PRs.

Reply:
We've been discussed in office and concluded that we will postpone it for a later phase, as @shurivich states that it would be very complicated to implement it. We will get System Tests session in the coming weeks from @shurivich that will explain in detail how the infrastructure is built. I envision that after this session we will recommend a few tweaks for the System Tests infra to support the use-case I presented above, alongside similar complex scenarios.
Resolving for now.